src/extension/inpage-script/index.js

```diff
@@ -1,86 +1,112 @@
-        paymentRequest = href.replace("lightning:", "");
-        link = lightningLink;
-      } else if (bitcoinLinkWithLighting) {
-        href = bitcoinLinkWithLighting.getAttribute("href").toLowerCase();
-        link = bitcoinLinkWithLighting;
-        const url = new URL(href);
-        const query = new URLSearchParams(url.search);
-        paymentRequest = query.get("lightning");
-      } else if (lnurlLink) {
-        href = lnurlLink.getAttribute("href").toLowerCase();
-        link = lnurlLink;
-        lnurl = href.replace(/^lnurl[pwc]:/i, "");
-      }
+        // if we did not find any paymentRequest and no LNURL we give up and return
+        if (!paymentRequest && !lnurl) {
+          return;
+        }
 
-      // if we did not find any paymentRequest and no LNURL we give up and return
-      if (!paymentRequest && !lnurl) {
-        return;
-      }
+        // it could be it is a LNURL behind a lightning: link
+        if (paymentRequest && paymentRequest.startsWith("lnurl")) {
+          lnurl = paymentRequest.replace(/^lnurl[pwc]:/i, ""); // replace potential scheme. the different lnurl types are handled in the lnurl action (by checking the type in the LNURL response)
+        }
+        // it could be a lightning address behind a lightning: link
+        if (paymentRequest && paymentRequest.match(/(\S+@\S+)/)) {
+          lnurl = paymentRequest.match(/(\S+@\S+)/)[1];
+        }
 
-      // it could be it is a LNURL behind a lightning: link
-      if (paymentRequest && paymentRequest.startsWith("lnurl")) {
-        lnurl = paymentRequest.replace(/^lnurl[pwc]:/i, ""); // replace potential scheme. the different lnurl types are handled in the lnurl action (by checking the type in the LNURL response)
-      }
-      // it could be a lightning address behind a lightning: link
-      if (paymentRequest && paymentRequest.match(/(\S+@\S+)/)) {
-        lnurl = paymentRequest.match(/(\S+@\S+)/)[1];
-      }
+        try {
+          await window.webln.enable();
+        } catch (e) {
+          console.error(e);
+        }
 
-      try {
-        await window.webln.enable();
-      } catch (e) {
-        console.error(e);
-      }
+        if (lnurl) {
+          try {
+            const response = await window.webln.lnurl(lnurl);
+            const responseEvent = new CustomEvent("lightning:success", {
+              bubbles: true,
+              detail: {
+                lnurl,
+                response,
+              },
+            });
+            link.dispatchEvent(responseEvent);
+          } catch (e) {
+            console.error(e);
+            if (
+              ![ABORT_PROMPT_ERROR, USER_REJECTED_ERROR].includes(e.message)
+            ) {
+              alert(`Error: ${e.message}`);
+            }
+          }
+        }
 
-      if (lnurl) {
         try {
-          const response = await window.webln.lnurl(lnurl);
+          const response = await window.webln.sendPayment(paymentRequest);
```

Comment:
these things here just changed because it was moved around, corret? 
we did not change anything on how the links are handled?

Reply:
resolved